service/worker/processor.go

```diff
@@ -188,20 +188,10 @@ func (p *replicationTaskProcessor) worker(workerWG *sync.WaitGroup) {
 						return p.process(msg)
 					}
 					err = backoff.Retry(op, replicationTaskRetryPolicy, p.isTransientRetryableError)
-					if err != nil {
-						// Check if this is an explicit ask to retry the task by handler
-						if _, ok := err.(*shared.RetryTaskError); ok {
-							// Increment the retryCount as we will retry the error upto ReplicatorMaxRetryCount before moving
-							// it to DLQ
-							retryCount++
-							time.Sleep(p.config.ReplicatorRetryDelay)
-							continue ProcessRetryLoop
-						}
-
+					if err != nil && p.isTransientRetryableError(err) {
 						// Keep on retrying transient errors for ever
-						if p.isTransientRetryableError(err) {
-							continue ProcessRetryLoop
-						}
+						time.Sleep(p.config.ReplicatorRetryDelay)
```

Comment:
Do we really need this extra delay now?  We already have a backoff on transient errors, and now RetryTaskError is also categorized as a transient error so I think we can get rid of this sleep interval.

Reply:
sure